pkg/redirectpolicy/manager.go

```diff
@@ -0,0 +1,531 @@
+		delete(rpm.frontends, frontend.Hash())
+	}
+}
+
+func (rpm *RedirectPolicyManager) updatePolicyConfigBackends(config *LRPConfig, isUpdate bool, backends ...backend) {
+	if isUpdate {
+		config.backends = append(config.backends, backends...)
+		rpm.policyConfigs[config.getPolicyID()] = config
+		rpm.notifyPolicyConfigUpdate(config)
+	} else {
+		// Currently, we expect number of LRP backends to be a single digit number.
+		// If this scales up, we might need to optimize this using sets.
+		var newBackends []backend
+		for _, currBk := range config.backends {
+			for _, removeBk := range backends {
+				if removeBk.StringWithProtocol() != currBk.StringWithProtocol() {
+					newBackends = append(newBackends, currBk)
+				}
+			}
+		}
+		config.backends = newBackends
+		rpm.notifyPolicyBackendDelete(config)
+	}
+}
+
+// Deletes service entry corresponding to the specified frontend from datapath.
+func (rpm *RedirectPolicyManager) notifyPolicyFrontendDelete(config *LRPConfig) {
+	if config.frontend == nil {
+		return
+	}
+	found, err := rpm.svcManager.DeleteService(*config.frontend)
+	delete(rpm.frontends, config.frontend.Hash())
+	config.frontend = nil
+	if !found || err != nil {
+		log.WithError(err).Debugf("Local redirect service for policy %s not deleted",
+			config.getPolicyID())
+	}
+}
+
+// Updates datapath with the deleted backend(s).
+func (rpm *RedirectPolicyManager) notifyPolicyBackendDelete(config *LRPConfig) {
+	if config.frontend.IP != nil {
+		if len(config.backends) > 0 {
+			rpm.upsertService(config)
+		} else {
+			// No backends so remove the service entry.
+			found, err := rpm.svcManager.DeleteService(*config.frontend)
+			if !found || err != nil {
+				log.WithError(err).Debugf("Local redirect service for policy %s not deleted",
+					config.getPolicyID())
+			}
+		}
+	}
+}
+
+// Checks if the given policy config is ready, and plumbs datapath with a service
+// entry created from the policy config.
+func (rpm *RedirectPolicyManager) notifyPolicyConfigUpdate(config *LRPConfig) {
+	if config.frontend.IP != nil && len(config.backends) > 0 {
+		// Policy config is ready.
+		rpm.upsertService(config)
+	}
+}
+
+// upsertService upserts a service entry for the given policy config that's ready.
+func (rpm *RedirectPolicyManager) upsertService(config *LRPConfig) {
+	frontendAddr := lb.L3n4AddrID{
+		L3n4Addr: *config.frontend,
+		ID:       lb.ID(0),
+	}
+	var backendAddrs []lb.Backend
+	for _, be := range config.backends {
+		backendAddrs = append(backendAddrs, lb.Backend{
+			NodeName: nodeTypes.GetName(),
+			L3n4Addr: be,
+		})
+	}
+	p := &lb.SVC{
+		Name:          config.name + localRedirectSvcStr,
+		Namespace:     config.namespace,
+		Type:          lb.SVCTypeLocalRedirect,
+		Frontend:      frontendAddr,
+		Backends:      backendAddrs,
+		TrafficPolicy: lb.SVCTrafficPolicyCluster,
+	}
+	log.Info("Upserting a service for redirect policy config %s with "+
+		"params %#v", config.getPolicyID(), p)
```

Comment:
I suspect that this is really developer-focused logging and is too verbose for `log.Info()`. As much as it can be nice to have this kind of information at runtime, if there's an error then we should be logging it below.